app/decorators/petition_search.rb

```diff
@@ -61,6 +65,15 @@ def execute_search_query
     end
   end
 
+  def execute_search_for_create_petition_query
+    Petition.search do |query|
```

Comment:
I agree with @otlaitil; as we have a new search method we can do the limit in there rather than using ``.take(3)`` on the results.  Ditto the facet thing, we almost certainly don't need it.